core/src/upgrade.rs

```diff
@@ -134,6 +135,36 @@ impl<T: AsRef<[u8]>> ProtocolName for T {
     }
 }
 
+/// Wrapper for printing a [`ProtocolName`] that is expected to be mostly ASCII
+///
+/// ```rust
+/// # use libp2p_core::upgrade::DisplayProtocolName;
+/// assert_eq!(DisplayProtocolName(b"/hello/1.0").to_string(), "/hello/1.0");
+/// assert_eq!(DisplayProtocolName("/hellö/").to_string(), "/hell<C3><B6>/");
+/// # assert_eq!(
+/// #     DisplayProtocolName((0u8..=255).collect::<Vec<_>>()).to_string(),
+/// #     (0..32).map(|c| format!("<{:02X}>", c))
+/// #         .chain((32..127).map(|c| format!("{}", char::from_u32(c).unwrap())))
+/// #         .chain((127..256).map(|c| format!("<{:02X}>", c)))
+/// #         .collect::<String>()
+/// # );
+/// ```
+pub struct DisplayProtocolName<N>(pub N);
```

Comment:
It would be nice if we could make this `pub(crate)`, otherwise it will be a breaking change to remove it later, despite it only being used internally.

However, I think this means we wouldn't be able to use the rustdoc test anymore.

Do you mind going back to a normal test?

Reply:
For that matter, it should probably live in `upgrade/apply.rs`.

Reply:
sure thing!